config/environments/production.rb

```diff
@@ -113,6 +113,8 @@
   # Uncomment and configure the following configuration option(s).
   # If your environment is more difficult (subdomains or different TLDs), try
   # https://github.com/synack/rack-allowed_hosts instead.
-  # config.action_controller.default_url_options = { host: "www.yoursite.com" }
-  # config.action_controller.asset_host = "www.yoursite.com"
+  # config.action_controller.default_url_options = {
```

Comment:
Do you want to assign `host_uri = URI.parse(ENV.fetch('HOST'))` instead of doing that step twice?

Reply:
Actually what I considered doing was having a global helper called `Rails.host_uri` (similar to `Rails.env` that would return the contents of `URI.parse(ENV.fetch('HOST'))`. It will be reused in several places and then you can do `Rails.host_uri.host` or `Rails.host_uri.to_s` depending on what you need. Thoughts?

Reply:
Not sure I'd want to monkey-patch it onto the Rails object, it could confuse people. One thing I could imagine as an alternative (and that I have done in similar scenarios) is adding an initializer that defines a module and wraps stuff, i.e. 

```ruby
module Constants
  HOST_URI = URI.parse(ENV.fetch('HOST'))
end
```

and whenever you need it you use `Constants::HOST_URI.host` which is just marginally more typing than `Rails.host_uri.host` but a lot clearer in communicating that this is our addition not some Rails feature.

Reply:
Love that. I think a `Constants` module make more sense...I'm not feeling a compelling need for a whole "config" module but constants are common. where do you normally put these?

Reply:
So would this be in `app/lib` or is there a best practice location for this? PG puts it in `lib` but rails 5 doesn't autoload that afaik and it seems like some people are moving to `app/lib` as a pattern but I haven't kept up with the recommendations there

Reply:
@hiattp Yup, my config module was for a multi tenanted app and also encapsulated the logic of accessing the right backends for different tenants etc. In the normal case `Constants` should be enough. As for the location, `app/lib` seems ok, though I'd also be fine with `config/initializers/constants.rb` since constants only should require a one-time setup on app boot. Anyway, your call.

We just need to make sure that we don't break other environments, so conditional definitions or submodules may be in order, i.e.

```ruby
module Constants
  modules Production
    HOST_URI = URI.parse(ENV.fetch('HOST'))
  end
end

# or

module Constants
  if Rails.env.production?
    HOST_URI = URI.parse(ENV.fetch('HOST')
  end
end
```

Reply:
I like the idea of putting it in an initializer. As for different submodules...why not just make these required in all environments and let the env do the job? That has the added benefit of signaling which local address to use, e.g. localhost:3000 or 5000 in most cases but for multitenanted apps xxx.lvh.me or whatever

Reply:
Yes, let's go for the easiest solution, i.e. what you said. 👍